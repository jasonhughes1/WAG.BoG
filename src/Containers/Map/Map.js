
import css from './Map.css';
import React, { Component } from 'react'
import GoogleMapReact from 'google-map-react'


const showMap = ({ text }) => <div>{ text }</div>;


export default class Map extends Component {
  static defaultProps = {
    center: { lat: 40.7446790, lng: -73.9485420 },
    zoom: 10
  }
render() {
    return (
      <div className='google-map'>
        <GoogleMapReact
          defaultCenter={ this.props.center }
          defaultZoom={ this.props.zoom }>
          <showMap
            lat={ 40.7473310 }
            lng={ -73.8517440 }
            text={ 'Where Waldo?' }
          />
        </GoogleMapReact>
      </div>
    )
  }
}
