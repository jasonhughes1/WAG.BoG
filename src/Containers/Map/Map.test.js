import React from 'react';
import { shallow } from 'enzyme';
import { Map } from './Map';

describe('Map tests', () => {
  let map;
  beforeEach( () => {
    const mockFunc = jest.fn();
    map = shallow(
      <Map
        dispatch={mockFunc}
        history={{key: 'value'}}
        location={['denver']}
        match={{key: 'value'}}
      />);
  });

  it('should match the snapshot', () => {
    expect(map).toMatchSnapshot();
  });

  it('Should be defined', () => {
    expect(map).toBeDefined();
  });
});
