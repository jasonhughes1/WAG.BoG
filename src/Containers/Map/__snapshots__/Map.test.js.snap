// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map tests should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Map
    dispatch={[Function]}
    history={
        Object {
            "key": "value",
          }
    }
    location={
        Array [
            "denver",
          ]
    }
    match={
        Object {
            "key": "value",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <GoogleMap
        debounced={true}
        defaultCenter={
                Object {
                        "lat": undefined,
                        "lng": undefined,
                      }
        }
        defaultZoom={10}
        distanceToMouse={[Function]}
        googleMapLoader={[Function]}
        hoverDistance={30}
        layerTypes={Array []}
        options={[Function]}
        style={
                Object {
                        "height": "100%",
                        "margin": 0,
                        "padding": 0,
                        "position": "relative",
                        "width": "100%",
                      }
        }
        yesIWantToUseGoogleMapApiInternals={false}
/>,
      "className": "google-map",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "debounced": true,
        "defaultCenter": Object {
          "lat": undefined,
          "lng": undefined,
        },
        "defaultZoom": 10,
        "distanceToMouse": [Function],
        "googleMapLoader": [Function],
        "hoverDistance": 30,
        "layerTypes": Array [],
        "options": [Function],
        "style": Object {
          "height": "100%",
          "margin": 0,
          "padding": 0,
          "position": "relative",
          "width": "100%",
        },
        "yesIWantToUseGoogleMapApiInternals": false,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <GoogleMap
          debounced={true}
          defaultCenter={
                    Object {
                              "lat": undefined,
                              "lng": undefined,
                            }
          }
          defaultZoom={10}
          distanceToMouse={[Function]}
          googleMapLoader={[Function]}
          hoverDistance={30}
          layerTypes={Array []}
          options={[Function]}
          style={
                    Object {
                              "height": "100%",
                              "margin": 0,
                              "padding": 0,
                              "position": "relative",
                              "width": "100%",
                            }
          }
          yesIWantToUseGoogleMapApiInternals={false}
/>,
        "className": "google-map",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "debounced": true,
          "defaultCenter": Object {
            "lat": undefined,
            "lng": undefined,
          },
          "defaultZoom": 10,
          "distanceToMouse": [Function],
          "googleMapLoader": [Function],
          "hoverDistance": 30,
          "layerTypes": Array [],
          "options": [Function],
          "style": Object {
            "height": "100%",
            "margin": 0,
            "padding": 0,
            "position": "relative",
            "width": "100%",
          },
          "yesIWantToUseGoogleMapApiInternals": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
